arch/arc/core/mpu/arc_mpu_v3_internal.h

```diff
@@ -845,6 +845,10 @@ static int arc_mpu_init(const struct device *arg)
 	/* Disable MPU */
 	arc_core_mpu_disable();
 
+	/* clear all the regions first */
```

Comment:
Please add your perfectly correct rationale in comments here so every reader of that code understands why do we do that. Though given we don't limit that fix-up to a particular board (which will add an extra `ifdef` which is not good) explain it in a more generic way (i.e. w/o mention of EMSK).

Reply:
Hi @abrodkin , thank you for the advice. Modified.